File: src/AutoRest.SdkExplorer/Model/Schema/SchemaEnum.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace AutoRest.SdkExplorer.Model.Schema
{
    public class SchemaEnum : SchemaBase
    {
        public const string SCHEMA_TYPE = "ENUM_SCHEMA";

        public List<SchemaEnumValue> Values { get; set; } = new List<SchemaEnumValue>();
        public string Description { get; set; } = string.Empty;

        private SchemaEnum()
            : base("OnlyForYamlDeserializor", SCHEMA_TYPE)
        {
        }

        [JsonConstructor]
        public SchemaEnum(string schemaKey)
            : base(schemaKey, SCHEMA_TYPE)
        {
        }

    }
}

File: src/AutoRest.SdkExplorer.TypeScript/src/Model/Schema/SchemaEnum.ts
import { SchemaBase } from "./SchemaBase";
import { SchemaEnumValue } from "./SchemaEnumValue";

export class SchemaEnum extends SchemaBase implements AutoRest.SdkExplorer.Interface.SchemaEnum {
    values: SchemaEnumValue[];
    description: string;


    constructor(data: AutoRest.SdkExplorer.Interface.SchemaEnum) {
        super(data)
        this.values = data.values!.map(d => new SchemaEnumValue(d));
        this.description = data.description ?? "";
    }

}
